### ISSUE TITLE
Twin Rotor Case

### ISSUE THREAD

**Issue by sreeraj37** [turbinesFoam/turbinesFoam#327]
Hello, Firstly thank you for the amazing repository.
Is it possible to extend the existing UNH-RVAT case (https://github.com/petebachant/UNH-RVAT-turbinesFoam) to a twin rotor case (not two turbines side by side), which means moving the center of rotor by some distance 1.2D using RAS? Is there any tutorials/cases available regarding twin rotors that i can refer to?

Thanks in Advance.

**Comment by petebachant** [comment]
There is no built-in support to define a multi-rotor turbine as a single one, but it should be equivalent to create two turbines near each other, no?

**Comment by sreeraj37** [comment]
Thank you so much for the reply.  

Yes it should be equivalent, to create two turbines side by side. Can you throw some lights in how to implement the same in cross flow case? 
Is it like adding one more _turbine_ in _fvOptions_ with a shifted origin say 1.5D along y direction (0 0 1.5*1), and setting _axis_ to _(0 0 -1)_ for counter rotating? Also how the forcing on second turbine can implemented? 

Thanks in Advance.

**Comment by petebachant** [comment]
Yep, that's basically it. Copy the existing turbine in `fvOptions`, then modify its location and axis. OpenFOAM makes it easy to copy entire subdictionaries or entries in the files using the `$` symbol so you don't literally need to copy/paste. See the OpenFOAM user guide for more info on that.

**Comment by sreeraj37** [comment]
Thanks for the guidance. I'm yet to explore the copying with the $ symbol, but i copied and edited the required entries in _fvOptions_ and _topoSetDict_ for making one more turbine at a distance 1.5D and making the second one rotate in the opposite direction( modified files are attached here). The simulation ran without any error but could find only one turbine in postprocessing with _paraview_. Wondering where exactly the issue might have occurred. Could you identify any glitches? 

[topoSetDict.txt](https://github.com/turbinesFoam/turbinesFoam/files/6873745/topoSetDict.txt)
[fvOptions.txt](https://github.com/turbinesFoam/turbinesFoam/files/6873747/fvOptions.txt)

**Comment by sreeraj37** [comment]
While placing one more turbine near by through _fvOption_ it's not adding the source term, I suppose there is no inbuilt ALM source addition available while extending the single rotor to twin rotor or to farm, right? 
For cross flow case I suppose I need to add source terms through _crossFlowTurbineALSource.C_ . am I in the right track?

**Comment by petebachant** [comment]
I see that the center of the `turbine2` `cellSet` doesn't coincide with the `origin` of `turbine2`. Why is that? Is the cell set being created properly in the mesh, and is the mesh large enough? Check the output logs for that step.

> For cross flow case I suppose I need to add source terms through _crossFlowTurbineALSource.C_

No, you should be able to run two turbines just fine without modifying the source code.

**Comment by sreeraj37** [comment]
Hi, thanks for the guidance. I could simulate the twin rotor case. It would be helpful if you could attend couple of quick questions.

1. How the domain independency study can be performed for a single turbine, as in how the specific values of x = 3.68m, y = 3.66m and z = 2.44m are selected for domain in the case: https://github.com/petebachant/UNH-RVAT-turbinesFoam ?

2. For the case https://github.com/petebachant/UNH-RVAT-turbinesFoam LES simulation, could Kres/(Kres + Ksgs) > 0.5 - 0.6 (ie 50% - 60% of the TKE being resolved by grid) be a good estimate of the LES grid  resolution? 


Thanks in Advance

**Comment by petebachant** [comment]
1. The domain was selected to match the experimental setup I used.
2. Unfortunately I didn't do that calculation so I'm not sure how much TKE is resolved.

**Comment by sreeraj37** [comment]
Hey, Thanks for the feedback and support being provided. About the twin turbine case I am simulating it would be helpful if you could attend couple of quick questions, 

1. I made the twin turbine case (with center distance 0.8D and both rotating in opposite sense) by adding one more turbine with a different origin in the  _fvOption_  and _topoSetDict_ files (attached herewith for reference).  The simulation was completed, but seeing the animations of velocity contours or vorticity magnitudes, it looks like the second turbine formed is a mirror image at the new origin, ie it is not only translated but also rotated as well. The vorticity magnitude animation is also attached for your reference.  Can you confirm my observation is right or not? Also in this case the turbine blades should collide, but it didn't throw any errors, does the code have a fool proofing about this collision?

2. What does the slip boundary condition used for velocity on the top  wall imply with reference to the tow tank experiment?


https://user-images.githubusercontent.com/79393969/148689895-7801b44f-0e25-4166-9d1e-b305ea3fc597.mp4
[topoSetDict.txt](https://github.com/turbinesFoam/turbinesFoam/files/7835343/topoSetDict.txt)
[fvOptions.txt](https://github.com/turbinesFoam/turbinesFoam/files/7835344/fvOptions.txt)
